### PR TITLE
gRPC + event bus(Kafka, Nats)

### PR DESCRIPTION
**Pull request by anastasiiakuzyk**

### CODE REVIEW

File: finesapp/src/main/kotlin/ua/anastasiia/finesapp/controller/grpc/GrpcService.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            .map { fine -> buildGetFineByCarPlateResponse(fine) }
+            .onErrorResume { exception -> handleCarPlateException(exception) }
+    }
+
+    override fun streamByCarPlate(request: Mono<StreamByCarPlateRequest>): Flux<StreamByCarPlateResponse> {
+        return request.flatMapMany { streamByCarPlateRequest ->
+            fineService.getFineByCarPlate(streamByCarPlateRequest.carPlate)
+                .flatMapMany { initFineState ->
+                    natsEventSubscriber.subscribe(streamByCarPlateRequest.carPlate)
+                        .map { trafficTicketAddedEvent ->
```
Comment (mykolaKhlopyk) [inline finesapp/src/main/kotlin/ua/anastasiia/finesapp/controller/grpc/GrpcService.kt:51]:
```
request.flatMap {
    streamByCarPlateRequest ->  fineService.getFineByCarPlate(streamByCarPlateRequest.carPlate)
}.flatMapMany {
    initFineState -> ...
```

File: finesapp/src/main/kotlin/ua/anastasiia/finesapp/controller/grpc/GrpcService.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                            buildStreamByCarPlateResponse(trafficTicketAddedEvent.fine.toFine().toResponse())
+                        }.startWith(
+                            buildStreamByCarPlateResponse(initFineState)
+                        )
+                }
+                .onErrorResume { exception -> handleCarPlateStreamException(exception) }
+        }
```
Comment (mykolaKhlopyk) [inline finesapp/src/main/kotlin/ua/anastasiia/finesapp/controller/grpc/GrpcService.kt:58]: NIT
 you combine two flux , but  natsEventSubscriber.subscribe(...) doesnt need t know about initFineState, so you can split these fluxes (maybe later relocate them to private methods)
```
return request.flatMapMany {
    streamByCarPlateRequest ->
    Flux.concat(
        fineService.getFineByCarPlate(streamByCarPlateRequest.carPlate).flatMap { initFineState ->
            buildStreamByCarPlateResponse(initFineState).toMono()
        },
        natsEventSubscriber.subscribe(streamByCarPlateRequest.carPlate)
            .map { trafficTicketAddedEvent ->
                buildStreamByCarPlateResponse(trafficTicketAddedEvent.fine.toFine().toResponse())
            }
    ).onErrorResume { exception -> handleCarPlateStreamException(exception) }
}
```

File: internal-api/src/main/proto/ua/anastasiia/finesapp/input/reqreply/trafficticket/add.proto [hunk 3]
```diff
@@ -11,7 +11,7 @@ message AddTrafficTicketRequest {
   commonmodels.trafficticket.TrafficTicket traffic_ticket = 2;
 }
 
-message AddTrafficTicketResponse {
+message AddTrafficTicketResponse{
```
Comment (mykolaKhlopyk) [inline internal-api/src/main/proto/ua/anastasiia/finesapp/input/reqreply/trafficticket/add.proto:14]: return space😂

**Review by ajax-gnatiuk-s — APPROVED** [review]

File: finesapp/src/main/kotlin/ua/anastasiia/finesapp/controller/nats/event/NatsEventSubscriber.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Sinks
+import ua.anastasiia.finesapp.NatsSubject
+import ua.anastasiia.finesapp.output.pubsub.trafficticket.TrafficTicketAddedEvent
+
+@Component
+class NatsEventSubscriber(private val connection: Connection) {
+
+    fun subscribe(carPlate: String): Flux<TrafficTicketAddedEvent> {
+        Sinks.many().multicast().onBackpressureBuffer<TrafficTicketAddedEvent>().apply {
```
[the remark this answers is not in the document]
Comment (ajax-gnatiuk-s) [reply finesapp/src/main/kotlin/ua/anastasiia/finesapp/controller/nats/event/NatsEventSubscriber.kt]: I'm ok with the both approaches

**Pull request merged**